pulsar-client-cpp/.clang-format

```diff
@@ -0,0 +1,20 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+BasedOnStyle: Google
+ColumnLimit: 90
```

Comment:
Probably we should try to get a style format that is close to most of current code base already. I haven't tried this yet but I'm curious at how much different it is! :D

Reply:
@merlimat @maskit This is the formatted codes by this settings.
https://github.com/Licht-T/incubator-pulsar/commit/7baaae96f39fd07f9e9768b9cb072dab7e39722a

Reply:
I'd like to at least stick to the 4 spaces indentation